src/OSPSuite.Presentation/Presenters/Importer/IModalImporterPresenter.cs

```diff
@@ -38,6 +53,31 @@ public ModalImporterPresenter(IModalImporterView view, IDataSetToDataRepositoryM
                {
                   var dataRepo = _dataRepositoryMapper.ConvertImportDataSet(d.DataSource, i++, pair.Key);
                   dataRepo.ConfigurationId = id;
+                  var molecule = dataRepo.ExtendedPropertyValueFor(moleculeName);
+                  var moleculeDescription = (metaDataCategories?.FirstOrDefault(md => md.Name == moleculeName)?.ListOfValues.FirstOrDefault(v => v.Key == dataRepo.ExtendedPropertyValueFor(moleculeName)))?.Value;
+                  var molecularWeightDescription = dataRepo.ExtendedPropertyValueFor(molWeightName);
+
+                  if (moleculeDescription != null)
+                  {
+                     if (string.IsNullOrEmpty(molecularWeightDescription))
+                     {
+                        molecularWeightDescription = moleculeDescription;
+                        dataRepo.ExtendedProperties.Add(new ExtendedProperty<string>() { Name = molWeightName, Value = moleculeDescription });
+                     }
+                     else
+                     {
+                        if (moleculeDescription != molecularWeightDescription)
```

Comment:
what is moleculeDescription? the name of the molecule? 
Why is this compared to the molecularWeight?

Reply:
Description is the value of the molecular weight coming either from molecule or directly from molecularWeight metadata